atlasdb-commons/src/test/java/com/palantir/common/compression/AbstractCompressionTests.java

```diff
@@ -0,0 +1,151 @@
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Random;
+
+import org.junit.After;
+import org.junit.Test;
+
+import com.google.common.io.ByteStreams;
+
+public abstract class AbstractCompressionTests {
+
+    private static final byte SINGLE_VALUE = 42;
+    private static final int BLOCK_SIZE = 1 << 16; // 64 KB
+
+    protected ByteArrayInputStream uncompressedStream;
+    protected InputStream compressingStream;
+    protected InputStream decompressingStream;
+
+
+
+
+    @Test
+    public void testEmptyStream() throws IOException {
+        initializeStreams(new byte[0]);
+        assertStreamIsEmpty(decompressingStream);
+    }
+
+    @Test
+    public void testSingleCharacterStream() throws IOException {
+        testStream_incompressible(1); // 1 byte input will always be incompressible
+    }
+
+    @Test
+    public void testSingleCharacterStream_singleByteRead() throws IOException {
+        byte[] uncompressedData = new byte[] { SINGLE_VALUE };
+        initializeStreams(uncompressedData);
+        int value = decompressingStream.read();
+
+        assertEquals(uncompressedData[0] & 0xFF, value);
+        assertStreamIsEmpty(decompressingStream);
+    }
+
+    @Test
+    public void testSingleBlock_compressible() throws IOException {
+        testStream_compressible(BLOCK_SIZE);
+    }
+
+    @Test
+    public void testSingleBlock_incompressible() throws IOException {
+        testStream_incompressible(BLOCK_SIZE);
+    }
+
+    @Test
+    public void testMultiBlock_compressible() throws IOException {
+        testStream_compressible(16 * BLOCK_SIZE);
+    }
+
+    @Test
+    public void testMultiBlock_incompressible() throws IOException {
+        testStream_incompressible(16 * BLOCK_SIZE);
+    }
+
+    @Test
+    public void testMultiBlock_singleByteReads() throws IOException {
+        byte[] uncompressedData = new byte[16 * BLOCK_SIZE];
+        fillWithIncompressibleData(uncompressedData);
+        initializeStreams(uncompressedData);
+
+        for (int i = 0; i < uncompressedData.length; ++i) {
+            int value = decompressingStream.read();
+            assertEquals(uncompressedData[i] & 0xFF, value);
+        }
+        assertStreamIsEmpty(decompressingStream);
+    }
+
+    @Test
+    public void testMultiBlock_readPastEnd() throws IOException {
+        byte[] uncompressedData = new byte[16 * BLOCK_SIZE];
+        fillWithCompressibleData(uncompressedData);
+        initializeStreams(uncompressedData);
+
+        byte[] decompressedData = new byte[17 * BLOCK_SIZE];
+        int bytesRead = ByteStreams.read(decompressingStream, decompressedData, 0, decompressedData.length);
+        assertEquals(uncompressedData.length, bytesRead);
+        assertArrayEquals(uncompressedData, Arrays.copyOf(decompressedData, bytesRead));
```

Comment:
instead of arrayEquals, assertArrayEquals and the other JUnit asserts, please can we use AssertJ's `assertThat` methods? Leads to better perf. `assertThat(decompressedData).startsWith(uncompressedData)` might well be possible.